stream-chat-android-state/src/main/java/io/getstream/chat/android/state/extensions/ChatClient.kt

```diff
@@ -152,6 +153,30 @@ public fun ChatClient.getRepliesAsState(
     return requestsAsState(coroutineScope).getReplies(messageId, messageLimit)
 }
 
+/**
+ * Returns thread replies in the form of [ThreadState], however, unlike [getRepliesAsState]
+ * it will return it only after the API call made to get replies has ended. Thread state
+ * will be returned regardless if the API call has succeeded or failed, the only difference is
+ * in how up to date the replies in the thread state are.
+ *
+ * @param messageId The ID of the original message the replies were made to.
+ * @param messageLimit The number of messages that will be initially loaded.
+ * @param coroutineScope The [CoroutineScope] used for executing the request.
+ *
+ * @return [ThreadState] wrapped inside a [Call].
+ */
+@InternalStreamChatApi
+@JvmOverloads
+public fun ChatClient.getRepliesAsStateCall(
+    messageId: String,
+    messageLimit: Int,
+    coroutineScope: CoroutineScope = CoroutineScope(DispatcherProvider.IO),
+): Call<ThreadState> {
+    return CoroutineCall(scope = state.scope) {
+        Result.Success(requestsAsState(coroutineScope).getRepliesAsStateCall(messageId, messageLimit))
+    }
+}
```

Comment:
Hey @MarinTolic,

Since it's an internal function, I'm wondering if we can make it `suspend` one.
```suggestion
public suspend fun ChatClient.getRepliesAsStateCall(
    messageId: String,
    messageLimit: Int,
): ThreadState {
    return requestsAsState(coroutineScope).getRepliesAsStateCall(messageId, messageLimit)
}
```

Reply:
Implemented as suggested. :D